app/common/lib/bookmarkFoldersUtil.js

```diff
@@ -117,11 +118,32 @@ const isMoveAllowed = (folderList, sourceDetail, destinationDetail) => {
   return true
 }
 
+const buildFolder = (folderDetails, folders) => {
+  folderDetails = makeImmutable(folderDetails)
+  let key = folderDetails.get('folderId')
+
+  if (!folderDetails.has('folderId')) {
+    key = getNextFolderId(folders)
+  }
+
+  return makeImmutable({
+    title: folderDetails.get('title'),
+    folderId: ~~key,
```

Comment:
jsperf is not working right now, so I can share specific numbers, but if this is less readable I can change it to `Number()` conversion. Not a problem